### DISCUSSION TITLE
Why does train.py build a val_dataset using train pipeline?

### DISCUSSION THREAD

**Discussion by gvalvano** [open-mmlab/mmgeneration#466] [Datasets] [answered]
Why do you use the train pipeline for the validation set at [this](https://github.com/open-mmlab/mmgeneration/blob/48c5b86a745d23c7f3a0310da966f2777b02634d/tools/train.py#L185) line of code?

To my understanding, the line of code:
```
val_dataset.pipeline = cfg.data.train.pipeline
```
should be 
```
val_dataset.pipeline = cfg.data.val.pipeline
```
Is there something I don't understand?

**Comment by LeoXing1996** [comment]
Hey @gvalvano, thanks for your interest in our project.
This seems a mistake, and we will fix this soon. We would also be glad that if you could create a pull request to fix this.

**Comment by gvalvano** [comment]
Hi @LeoXing1996 Thanks for the help. Sure, I'll be happy to do so!

**Comment by gvalvano** [comment, marked as answer]
Opened pull request #244

**Comment by gvalvano** [comment]
Hi @LeoXing1996 it seems that the pull request failed during the "lint" process. I don't have familiarity with this. Should I take any action to address this issue?

**Comment by LeoXing1996** [comment]
Please feel free. That lint error is caused by the network error. I just rerun the CI and all checks are passed. We will merge this in this version.
